src/main/java/org/ibp/api/java/impl/middleware/dataset/validator/StudyValidator.java

```diff
@@ -50,14 +50,8 @@ public void validate(final Integer studyId, final Boolean shouldBeUnlocked) {
 			throw new ResourceNotFoundException(this.errors.getAllErrors().get(0));
 		}
 
-		final WorkbenchUser loggedInUser = this.securityService.getCurrentlyLoggedInUser();
-
-		if (shouldBeUnlocked
-			&& study.isLocked()
-			&& !study.getCreatedBy().equals(loggedInUser.getUserid().toString())
-			&& !loggedInUser.isSuperAdmin()) {
-			this.errors.reject("study.is.locked", "");
-			throw new ForbiddenException(this.errors.getAllErrors().get(0));
+		if (shouldBeUnlocked) {
+			this.checkIfStudyIsUnlocked(study);
```

Comment:
checkIfStudyIsUnlocked is inaccurate because the user can still edit a locked study if he is the owner or is a superadmin